packages/core/src/node/runtimeModule/siteData/index.ts

```diff
@@ -28,7 +28,8 @@ function deletePriviteKey<T>(obj: T): T {
 
 export async function siteDataVMPlugin(context: FactoryContext) {
   const { config, alias, userDocRoot, routeService, pluginDriver } = context;
-  const userConfig = config;
+  // deep clone to avoid unexpected changes to objects
+  const userConfig = JSON.parse(JSON.stringify(config));
```

Comment:
Will the config contain some properties that can not be serialized?

Reply:
such as RegExp or function

Reply:
see the latest code, just use Object.assign to prevent the search object being modified

Reply:
LGTM